### ISSUE TITLE
[Question]: Is route interception on a page supposed to work in every case (Chromium)?

### ISSUE THREAD

**Issue by localnerve** [microsoft/playwright#7588]
### Your question

Is Page.route supposed to work on page requests alongside a service worker in every case?

I found a reproducible case where this doesn't work. Not sure if this is a bug, just not documented, or I missed it.

After troubleshooting and observation (and finding a workaround), I went back and created a simple, minimal example where Page.route (for page requests), alongside a service worker, will hang a test indefinitely.

Here are the conditions required for trouble:
* Chromium, as installed by Playwright
* MacOs, Catalina
1. Page.route called before any navigation for complete capture
2. Page.click called on [type='submit'] to trigger form, handled with client fetch as 'application/json'
3. Expressjs (4.17.1 bare-bones log and reply) on http://localhost (`localhost` important, perhaps protocol too but **I didn't check proto** - https remote host works)
4. A service worker (workbox 6.1.5) installed, `NetworkOnly` (no cache) strategy on the route POST

The observed failure:
In the failure condition, the request is still made (as logged by Expressjs) to localhost, but any `Page.on('response', handler)` or `Page.route(() => true, handler)` or matching `Page.waitForResponse` will never execute.

The workaround for me is to avoid Page.route or stop the service worker from installing.

I debugged with `pw:api` and all combos of Page.click options. This led me to code the minimal reproducible example, and ask the question.

In the minimal reproduction:
When Page.route, Page.click, and localhost server are used, adding or removing the service worker toggles the failure condition.

**Comment by aslushnikov** [comment]
> Is Page.route supposed to work on page requests alongside a service worker in every case?

@localnerve page.route doesn't work for service worker routes. It intercepts only **network** traffic from the page. We have an issue filed for this: https://github.com/microsoft/playwright/issues/1090 Feel free to upvote!

For now, we'd recommend disabling service workers on the website. 

P.S. I appreciate all the debugging you did! Let me fix the docs so that this behavior won't be surprising in future.

**Comment by localnerve** [comment]
The phrase, "service worker routes" is confusing. The request is initiated from the page. When I read "service worker route", I presume a request initiated by the service worker.

Exceptions/Outliers:
1. Routes other than the POST initiated by the page fetch work.
2. Works with remote host over https, not localhost http.

Feels like we are missing the core of the underlying issue.

**Comment by aslushnikov** [comment]
> The phrase, "service worker routes" is confusing. The request is initiated from the page. When I read "service worker route", I presume a request initiated by the service worker.

My bad, let me rephrase: it won't intercept requests intercepted by service worker.

**Comment by localnerve** [comment]
That makes sense. Thanks for clearing this up. Please consider using that phraseology, or making that distinction, in the docs.
